azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -88,7 +88,8 @@ def decrypt_p7m(self, p7m_file, trans_prv_file, trans_cert_file, pem_file):
 
             first_proc = subprocess.Popen(first_cmd, stdout=subprocess.PIPE)
 
-            second_proc = subprocess.Popen(second_cmd, stdin=first_proc.stdout, stdout=subprocess.PIPE)
+            second_proc = subprocess.Popen(second_cmd, stdin=first_proc.stdout,
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
A weird question. The default value is `stderr=None`. If it was None, how were we getting the weird output message in the console? Any idea?

Reply:
My understanding is that `None` defaults to passing through the output to the agent process.

Reply:
Ahh you're right - 

>stdout=None means that subprocess prints to whatever place your script prints. stdout=PIPE means that subprocess' stdout is redirected to a pipe that you should read e.g., using process.communicate() to read all at once or using process.stdout object to read via a file/iterator interfaces